File: Caesar/caesar.py
# Implement Implement a program that encrypts messages using Caesar’s cipher, per the below.

# $ python caesar.py 13
# plaintext:  HELLO
# ciphertext: URYYB


from sys import argv
from cs50 import get_string


def main():
    # Verify if input data is valid
    if len(argv) != 2:
        print("Usage: python caesar.py integer")
        exit(1)

    # Get key
    key = int(argv[1])
    print(ord('a'))

    # Get plaintext
    plaintext = get_string("plaintext: ")

    # Encipher and print the ciphertext
    z = 0
    print("ciphertext: ", end="")

    for i in range(len(plaintext)):
        if plaintext[i] >= 'a' and plaintext[i] <= 'z':
            z = chr((((ord(plaintext[i]) + key - 97) % 26) + 97))
            print(z, end="")
        elif plaintext[i] >= 'A' and plaintext[i] <= 'Z':
            z = chr((((ord(plaintext[i]) + key - 65) % 26) + 65))
            print(z, end="")
        else:
            print(plaintext[i], end="")

    # Print newline
    print()


main()

File: Caesar-C/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        string s = 0;
        for (int i = 0, n = strlen(argv[1]); i < n; i++)
        {
            if (argv[1][i] >= '0' && argv[1][i] <= '9')
            {

            }
            else
            {
                printf("Usage: ./caesar key\n");
                return 1;
            }
        }
        int k = atoi(argv[1]);
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    string p = get_string("plaintext: ");
    int z = 0;
    int k = atoi(argv[1]);
    printf("ciphertext: ");    
    for (int i = 0, l = strlen(p); i < l; i++)
    {
        if (p[i] >= 'a' && p[i] <= 'z')
        {
            z = ((((int) p[i] + k - 97) % 26) + 97);
            printf("%c", z);
        }
        else if (p[i] >= 'A' && p[i] <= 'Z')
        {
            z = ((((int) p[i] + k - 65) % 26) + 65);
            printf("%c", z);
        }
        else
        {
            printf("%c", p[i]);
        }
    }
    printf("\n");
}
